compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/XCLBinGen.cpp

```diff
@@ -1029,17 +1028,26 @@ struct RemoveAlignment2FromLLVMLoadPass
 }  // namespace
 
 static LogicalResult generateUnifiedObject(
-    MLIRContext *context, ModuleOp moduleOp, const std::string &outputFile,
+    MLIRContext *context, AIE::DeviceOp deviceOp, const std::string &outputFile,
     bool printIRBeforeAll, bool printIRAfterAll, bool printIRModuleScope,
     bool timing, bool useChess, bool verbose, Path tempDir,
     std::optional<Path> vitisDir, const std::string &targetArch,
     Path peanoDir) {
-  PassManager pm(context, moduleOp.getOperationName());
+  // TODO(newling) to avoid nesting the DeviceOp in a ModuleOp,
+  // we need to make changes to core-to-standard-pass.
+  ModuleOp moduleWithOneDevice;
+  {
+    OpBuilder opBuilder(deviceOp.getContext());
+    moduleWithOneDevice = opBuilder.create<ModuleOp>(deviceOp.getLoc());
+    opBuilder.setInsertionPointToStart(moduleWithOneDevice.getBody());
+    opBuilder.clone(*deviceOp.getOperation());
+  }
+  PassManager pm(context, moduleWithOneDevice.getOperationName());
```

Comment:
yea I'd prefer that were done here - is it a lot of work?

Reply:
Moving the ModuleOp creation into core-to-standard pass is easy (I've done it locally). The problem is that the DeviceOp gets destroyed in the core-to-standard-pass, and so if we move the ModuleOp creation into core-to-standard pass, we lose a handle on any op. Later passes run on a ModuleOp. I could hackily _not_ remove the DeviceOp in core-to-standard, just so that we can get it's parent ModuleOp later, but I don't think that's any cleaner than what we have here. 

I think there's still along way down to go in simplifying the overall flow  -- can we for example just never create a DeviceOp, just use a ModuleOp, seems to have everything we need -- I'd prefer to have a crack at this in a later PR.

Reply:
I mean this is more reason to have just left module op everywhere but at this point no need to revert all your changes here - just leave this hackish thing.

> I think there's still along way down to go in simplifying the overall flow -- can we for example just never create a DeviceOp, just use a ModuleOp, seems to have everything we need -- I'd prefer to have a crack at this in a later PR.

keep pondering this and you'll come to the same conclusion I did: we don't need _any_ of the AIE ops :)

Reply:
Not having a ModuleOp as the root is probably asking for trouble, I'm going to make some changes..,

Reply:
see https://github.com/nod-ai/iree-amd-aie/blob/6a0c637c9da26022aa6f094e3496b6982b885c79/compiler/plugins/target/AMD-AIE/aie/test/aiert_insts.mlir#L10-L16